test/UnbrandedProjects/NoTest-TypeSpec/src/Generated/Internal/ClientUriBuilder.cs

```diff
@@ -42,7 +42,7 @@ public void AppendPath(string value, bool escape)
                 value = Uri.EscapeDataString(value);
             }
 
-            if (value[0] == PathSeparator)
+            if (value[0] == PathSeparator && PathBuilder.Length > 0 && PathBuilder[PathBuilder.Length - 1] == PathSeparator)
```

Comment:
Would something like:

```csharp
        if (PathBuilder[PathBuilder.Length - 1] == '/' && value[0] == '/')
        {
            PathBuilder.Remove(PathBuilder.Length - 1, 1);
        }
```

I had started drafting some changes to this in https://github.com/Azure/azure-sdk-for-net/pull/43436, to implement test coverage I was thinking we could eventually move into the generator code base.

The reason to remove from the builder instead of calling value.Substring is that substring creates a new string instance which makes an allocation.

Reply:
Yeah I know that we should avoid of using substring - my original plan is to replace it with `ReadOnlySpan.Slice` but I think your solution is simpler, I will change the code to this.

Reply:
BTW do we need to check `PathBuilder.Length > 0` first? Or this is somehow guaranteed always to be true?